interpreter/ruby/ruby.go

```diff
@@ -808,26 +828,57 @@ func Loader(ebpf interpreter.EbpfHandler, info *interpreter.LoaderInfo) (interpr
 	if version < rubyVersion(3, 0, 0) {
 		currentCtxSymbol = "ruby_current_execution_context_ptr"
 	}
-	currentCtxPtr, err := ef.LookupSymbolAddress(currentCtxSymbol)
-	if err != nil {
-		return nil, fmt.Errorf("%v not found: %v", currentCtxSymbol, err)
-	}
+
+	var currentEcTlsOffset libpf.SymbolValue
+	var interpRanges []util.Range
 
 	// rb_vm_exec is used to execute the Ruby frames in the Ruby VM and is called within
 	// ruby_run_node  which is the main executor function since Ruby v1.9.0
 	// https://github.com/ruby/ruby/blob/587e6800086764a1b7c959976acef33e230dccc2/main.c#L47
-	symbolName := libpf.SymbolName("rb_vm_exec")
+	interpSymbolName := libpf.SymbolName("rb_vm_exec")
 	if version < rubyVersion(2, 6, 0) {
-		symbolName = libpf.SymbolName("ruby_exec_node")
+		interpSymbolName = libpf.SymbolName("ruby_exec_node")
 	}
-	interpRanges, err := info.GetSymbolAsRanges(symbolName)
+
+	var currentEcSymbol *libpf.Symbol
+	currentEcSymbolName := libpf.SymbolName("ruby_current_ec")
+
+	log.Infof("Ruby %d.%d.%d detected, looking for currentCtxPtr=%q, currentEcSymbol=%q",
+		(version>>16)&0xff, (version>>8)&0xff, version&0xff, currentCtxSymbol, currentEcSymbolName)
+
+	currentCtxPtr, err := ef.LookupSymbolAddress(currentCtxSymbol)
 	if err != nil {
-		return nil, err
+		log.Debugf("Direct lookup of %v failed: %v, will try fallback", currentCtxSymbol, err)
 	}
 
+	interpRanges, err = info.GetSymbolAsRanges(interpSymbolName)
+	if err != nil {
+		log.Debugf("Direct lookup of %v failed: %v, will try fallback", interpSymbolName, err)
+	}
+
+	err = ef.VisitUntilSymbol(func(s libpf.Symbol) bool {
+		if s.Name == currentEcSymbolName {
+			currentEcSymbol = &s
+		}
+		if len(interpRanges) == 0 && s.Name == interpSymbolName {
+			interpRanges = info.SymbolAsRanges(&s)
+		}
+		if len(interpRanges) > 0 && currentEcSymbol != nil {
+			return false
+		}
+		return true
+	})
+
+	if currentEcSymbol != nil {
+		currentEcTlsOffset = currentEcSymbol.Address
+	}
+
+	log.Infof("Discovered EC %x, interp ranges: %v", currentEcTlsOffset, interpRanges)
```

Comment:
```suggestion
	log.Debugf("Discovered EC %x, interp ranges: %v", currentEcTlsOffset, interpRanges)
```

Reply:
fix in https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/710/commits/e51060992e57e4dc2143093b71fd045cf72257b9